R/Driver.R

```diff
@@ -61,6 +62,8 @@ setMethod(
 #' name for the OdbcConnect object returned from `dbConnect()`. However if the
 #' driver does not return a valid value it can be set manually with this
 #' parameter.
+#' @param attributes An S4 object of connection attributes that are passed
```

Comment:
Why make this an S4 object and not (e.g.) just a named list?

Reply:
It seemed a bit awkward housing connection attributes that may be specific to individual DBs in the same structure.
Without needing dispatch, I probably would not have used S4, but its already all over DBI so it seemed OK.

Happy to go a different route if you think that's an overkill.

Reply:
Ooooh, this isn't a generic way to pass name-value pairs to the driver, it's specifically tailored to solve the Azure issue?

Reply:
Sorry for the confusion.

It is intended to pass name-value pairs to the driver.  However these come in different flavors.  Some are generic  ODBC attributes ( timeout, catalog, etc ) that are (expected to be) implemented across all back-ends, and in time we can support these through this interface; they would fit naturally as slots of the parent ```ConnectionAttributes``` object.  However there may be other attributes that are back-end specific.  The azure token is one of them.

Reply:
Hmmm, in that case I still think it would be slightly easier to provide a named list. You make it generic by providing a an `odbc_serialize()` generic which you'd apply to each element of the list. And for now you'd just start with `AzureToken`. Does that make sense?

Reply:
Yeah, that sounds good.

Reply:
Thanks / should be done; let me know if I missed anything.

Pipeline looks to be failing, but I believe it is for an unrelated reason, [fixed here.](https://github.com/r-dbi/odbc/pull/522)